app/services/stats/management_information/queries/base_count_query.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Stats
+  module ManagementInformation
+    class BaseCountQuery
+      include JourneyQueryable
+      include ClaimTypeQueryable
+
+      attr_reader :scheme
+
+      def self.call(**kwargs)
+        new(kwargs).call
+      end
+
+      def initialize(day:, date_column_filter:)
+        @day = day
+        @date_column_filter = sql_quote(date_column_filter)
+        raise ArgumentError, 'day must be provided' if @day.blank?
+      end
+
+      def call
+        prepare
+        ActiveRecord::Base.connection.execute(query)
+      end
+
+      private
+
+      def prepare
+        ActiveRecord::Base.connection.execute(drop_journeys_func)
+        ActiveRecord::Base.connection.execute(create_journeys_func)
+      end
+
+      def sql_quote(column_name)
+        ActiveRecord::Base.connection.quote_column_name(column_name)
+      end
+
+      def query
+        raise RuntimeError
+      end
```

Comment:
I'm not sure this is required. If this method is missing in a child then a `NoMethodError` exception will be raise, which gives more information than `RuntimeError`.

Reply:
i believe if the method is missing in the child/subclass then the superclasses method will be called, no?

Reply:
This is the top level so there is no superclass. This will raise a `RuntimeError` exception, which I think is less helpful than the `NoMethodError` exception that would be raised if this method doesn't exist.

Reply:
I could change to NoMethodError but having the method in the superclass is for documentary purposes as typical for template method pattern. So if you define a subclass and do not define a query method and call it you get RunTimeEror. You should never be used this abstract class, which i could define as such too.